store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/retriever/AccumuloRetriever.java

```diff
@@ -29,39 +29,45 @@
 import uk.gov.gchq.gaffer.accumulostore.key.AccumuloElementConverter;
 import uk.gov.gchq.gaffer.accumulostore.key.IteratorSettingFactory;
 import uk.gov.gchq.gaffer.accumulostore.key.RangeFactory;
-import uk.gov.gchq.gaffer.commonutil.iterable.CloseableIterable;
-import uk.gov.gchq.gaffer.commonutil.iterable.CloseableIterator;
+import uk.gov.gchq.gaffer.commonutil.CloseableUtil;
 import uk.gov.gchq.gaffer.data.element.Element;
 import uk.gov.gchq.gaffer.data.element.function.ElementFilter;
 import uk.gov.gchq.gaffer.data.element.function.ElementTransformer;
+import uk.gov.gchq.gaffer.data.elementdefinition.view.View;
 import uk.gov.gchq.gaffer.data.elementdefinition.view.ViewElementDefinition;
+import uk.gov.gchq.gaffer.operation.Operation;
 import uk.gov.gchq.gaffer.operation.graph.GraphFilters;
 import uk.gov.gchq.gaffer.store.StoreException;
 import uk.gov.gchq.gaffer.user.User;
 
+import java.io.Closeable;
+import java.util.Iterator;
 import java.util.Set;
 
-public abstract class AccumuloRetriever<OP extends Output & GraphFilters, O_ITEM> implements CloseableIterable<O_ITEM> {
+public abstract class AccumuloRetriever<O_ITEM> implements Iterable<O_ITEM>, Closeable, GraphFilters {
+
     private static final Logger LOGGER = LoggerFactory.getLogger(AccumuloRetriever.class);
 
-    protected CloseableIterator<O_ITEM> iterator;
+    protected Iterator<O_ITEM> iterator;
     protected final AccumuloStore store;
     protected final Authorizations authorisations;
     protected final User user;
     protected final RangeFactory rangeFactory;
     protected final IteratorSettingFactory iteratorSettingFactory;
-    protected final OP operation;
+    protected final Operation operation;
     protected final AccumuloElementConverter elementConverter;
     protected final IteratorSetting[] iteratorSettings;
+    protected final View view;
 
-    protected AccumuloRetriever(final AccumuloStore store, final OP operation,
+    protected AccumuloRetriever(final AccumuloStore store, final Operation operation, final View view,
                                 final User user, final IteratorSetting... iteratorSettings)
             throws StoreException {
         this.store = store;
         this.rangeFactory = store.getKeyPackage().getRangeFactory();
         this.iteratorSettingFactory = store.getKeyPackage().getIteratorFactory();
         this.elementConverter = store.getKeyPackage().getKeyConverter();
         this.operation = operation;
```

Comment:
I wonder if this field is needed `Operation`

Reply:
Yeah, it's not needed since I've added `View` as a parameter. Have removed